src/coordinator/ts/values_test.go

```diff
@@ -0,0 +1,114 @@
+package ts
+
+import (
+	"testing"
+	"time"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"math"
+	"fmt"
+)
+
+type testSample struct {
+	input       Datapoints
+	expected    []float64
+	start       time.Time
+	end         time.Time
+	interval    time.Duration
+	hasNans     bool
+	description string
+}
+
+func generateDatapoints(start time.Time, interval time.Duration, num int) Datapoints {
+	dps := make(Datapoints, num)
+	t := start
+	for i := 0; i < num; i++ {
+		dps[i] = Datapoint{
+			Timestamp: t,
+			Value:     float64(i),
+		}
+		t = t.Add(interval)
+	}
+	return dps
+}
+func createExamples() []testSample {
+	now := time.Time{}
+	samples := []testSample{
+		{
+			input:       generateDatapoints(now.Add(-5*time.Second), time.Second, 5),
+			expected:    []float64{0, 1, 2, 3, 4,},
```

Comment:
nit: Don't need the trailing `,` in the array if it's on a single line

Reply:
yeah, I personally like to make it explicit.